dubbo-common/src/main/java/org/apache/dubbo/common/utils/StringUtils.java

```diff
@@ -478,10 +477,23 @@ public static boolean isContains(String[] values, String value) {
         return false;
     }
 
-    public static boolean isNumeric(String str) {
-        if (str == null) {
+    public static boolean isNumeric(String str, boolean allowDot) {
+        if (str == null || str.isEmpty()) {
             return false;
         }
+        if (allowDot) {

```

Comment:
Can handle something like ".1" or not?

Reply:
StringUtils.isNumeric(left, false) && StringUtils.isNumeric(right, false)

Reply:
That's no problem, it will go to the logic below and it will return false.

Reply:
`Double.valueOf(".1"))` will produce 0.1, so I personally think it's better to support this pattern.